### PR TITLE
add cluster name into the infra and workload machinesets name to iden…

### PR DESCRIPTION
**Pull request by qiliRedHat**
…tify the resource owner on provider side

### Description
https://github.com/cloud-bulldozer/scale-ci-deploy/issues/170

### Fixes
Add cluster name into the name of infra and workload machineset to help identify the resource from provider side.

### CODE REVIEW

**Comment by qiliRedHat** [comment]
@mohit-sheth Please help to review. How to test the change before merge?

**Comment by kedark3** [comment]
@qiliRedHat thanks for this PR - I would definitely encourage testing this before merge - as some cloud providers have limit on length of names etc. Don't want creation of these nodes to fail with long names.

**Review by kedark3 — APPROVED** [review]

File: OCP-4.X/roles/post-install/templates/gcp-infra-node-machineset.yml.j2 [hunk 1]
```diff
@@ -78,15 +78,15 @@ items:
         {{machineset_metadata_label_prefix}}/cluster-api-cluster: {{cluster_name.stdout}}
         {{machineset_metadata_label_prefix}}/cluster-api-machine-role: infra
         {{machineset_metadata_label_prefix}}/cluster-api-machine-type: infra
-        {{machineset_metadata_label_prefix}}/cluster-api-machineset: infra-b
+        {{machineset_metadata_label_prefix}}/cluster-api-machineset: infra-{{cluster_name.stdout}}-a
```
Comment (mohit-sheth) [inline OCP-4.X/roles/post-install/templates/gcp-infra-node-machineset.yml.j2:81]: s/a/b ?

File: OCP-4.X/roles/post-install/templates/gcp-infra-node-machineset.yml.j2 [hunk 2]
```diff
@@ -78,15 +78,15 @@ items:
...
     template:
       metadata:
         creationTimestamp: null
         labels:
           {{machineset_metadata_label_prefix}}/cluster-api-cluster: {{cluster_name.stdout}}
           {{machineset_metadata_label_prefix}}/cluster-api-machine-role: infra
           {{machineset_metadata_label_prefix}}/cluster-api-machine-type: infra
-          {{machineset_metadata_label_prefix}}/cluster-api-machineset: infra-b
+          {{machineset_metadata_label_prefix}}/cluster-api-machineset: infra-{{cluster_name.stdout}}-a
```
Comment (mohit-sheth) [inline OCP-4.X/roles/post-install/templates/gcp-infra-node-machineset.yml.j2:89]: s/a/b ?

File: OCP-4.X/roles/post-install/templates/gcp-infra-node-machineset.yml.j2 (hunk 1 above)
Comment (qiliRedHat) [reply OCP-4.X/roles/post-install/templates/gcp-infra-node-machineset.yml.j2]: Thanks Mohit! Corrected the typo.

File: OCP-4.X/roles/post-install/templates/gcp-infra-node-machineset.yml.j2 (hunk 2 above)
Comment (qiliRedHat) [reply OCP-4.X/roles/post-install/templates/gcp-infra-node-machineset.yml.j2]: Thanks Mohit! Corrected the typo.

**Comment by qiliRedHat** [comment]
Talked with @mohit-sheth, he explained why the cluster name is not in the machinset.

> we initially removed the cluster name from the machineset bc if the cluster name is long enough the machineset name exceeds the length and openshift won't create it

He made a change to fix the disks that are hanging around even after cleanup https://github.com/cloud-bulldozer/scale-ci-deploy/pull/172

As #172 already addressed the storage deletion issue, there won't be infra or workload storage resource left, and considering this PR will bring back the risk of 'machineset name exceeds the length', I think this PR can be closed for now.

**Comment by mohit-sheth** [comment]
LGTM

**Comment by qiliRedHat** [comment]
Talked to To Huang in QE team -export for GCP and installer,  and then talked with @mohit-sheth, this fix is supposed to fix the firewall issue as well. So we decided to take it. I'll keep an eye on long-cluster name issue.

**Pull request merged**